packages/multichain-network-controller/src/MultichainNetworkController.ts

```diff
@@ -0,0 +1,204 @@
+    state,
+  }: {
+    messenger: MultichainNetworkControllerMessenger;
+    state?: Omit<
+      Partial<MultichainNetworkControllerState>,
+      'multichainNetworkConfigurationsByChainId'
+    >;
+  }) {
+    super({
+      messenger,
+      name: MULTICHAIN_NETWORK_CONTROLLER_NAME,
+      metadata: MULTICHAIN_NETWORK_CONTROLLER_METADATA,
+      state: {
+        ...getDefaultMultichainNetworkControllerState(),
+        ...state,
+      },
+    });
+
+    this.#subscribeToMessageEvents();
+    this.#registerMessageHandlers();
+  }
+
+  /**
+   * Sets the active EVM network.
+   *
+   * @param id - The client ID of the EVM network to set active.
+   */
+  async #setActiveEvmNetwork(id: NetworkClientId): Promise<void> {
+    // Notify listeners that setActiveNetwork was called
+    this.messagingSystem.publish(
+      'MultichainNetworkController:networkDidChange',
+      id,
+    );
+
+    // Indicate that the non-EVM network is not selected
+    this.update((state) => {
+      state.isEvmSelected = true;
+    });
+
+    // Prevent setting same network
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+
+    if (id === selectedNetworkClientId) {
+      // EVM network is already selected, no need to update NetworkController
+      return;
+    }
+
+    // Update evm active network
+    await this.messagingSystem.call('NetworkController:setActiveNetwork', id);
+  }
+
+  /**
+   * Sets the active non-EVM network.
+   *
+   * @param id - The chain ID of the non-EVM network to set active.
+   */
+  #setActiveNonEvmNetwork(id: SupportedCaipChainId): void {
+    if (id === this.state.selectedMultichainNetworkChainId) {
+      if (!this.state.isEvmSelected) {
+        // Same non-EVM network is already selected, no need to update
+        return;
+      }
+
+      // Indicate that the non-EVM network is selected
+      this.update((state) => {
+        state.isEvmSelected = false;
+      });
+
+      // Notify listeners that setActiveNetwork was called
+      this.messagingSystem.publish(
+        'MultichainNetworkController:networkDidChange',
+        id,
+      );
+    }
+
+    // Notify listeners that setActiveNetwork was called
+    this.messagingSystem.publish(
+      'MultichainNetworkController:networkDidChange',
+      id,
+    );
+
+    this.update((state) => {
+      state.selectedMultichainNetworkChainId = id;
+      state.isEvmSelected = false;
+    });
+  }
+
+  /**
+   * Sets the active network.
+   *
+   * @param id - The non-EVM Caip chain ID or EVM client ID of the network to set active.
+   * @returns - A promise that resolves when the network is set active.
+   */
+  async setActiveNetwork(
+    id: SupportedCaipChainId | NetworkClientId,
+  ): Promise<void> {
+    if (isCaipChainId(id)) {
+      const isSupportedCaipChainId = checkIfSupportedCaipChainId(id);
+      if (!isSupportedCaipChainId) {
+        throw new Error(`Unsupported Caip chain ID: ${id}`);
```

Comment:
A `@typescript-eslint` rule is reporting that `id` is `never` here (which I think is right). There might be another way to solve this but in the meantime to appease linting maybe best to stringify it? It should be safe since we know realistically it's still a string:

``` suggestion
        throw new Error(`Unsupported Caip chain ID: ${String(id)}`);
```